ExampleExtractor/Program.cs

```diff
@@ -37,20 +39,40 @@ protected override async Task Start()
     }
 }
 
+// Class for flat command line arguments
+class Options
+{
+    [CommandLineOption("Specify command line argument", true, "-c")]
+    public string ConfigPath { get; set; }
+}
+
 // Then, in the Main() method:
 class Program
 {
-    static async Task Main()
+    static async Task Main(string[] args)
     {
-        await ExtractorRunner.Run<BaseConfig, MyExtractor>(
-            "config.yml",
-            new[] { 1 },
-            "my-extractor",
-            "myextractor/1.0.0",
-            false,
-            true,
-            true,
-            true,
-            CancellationToken.None).ConfigureAwait(true);
+        var command = new RootCommand
+        {
+            Description = "Simple example extractor"
+        };
+        var binder = new AttributeBinder<Options>();
+        binder.AddOptionsToCommand(command);
+
+        command.SetHandler<Options>(async opt =>
+        {
+            // This can also be invoked directly in main, to not have a CLI.
+            await ExtractorRunner.Run<BaseConfig, MyExtractor>(
+                opt.ConfigPath ?? "config.yml",
+                new[] { 1 },
+                "my-extractor",
+                "myextractor/1.0.0",
+                false,
+                true,
+                true,
+                true,
```

Comment:
What is the best practice for stuff like this in C#? A long list of parameters is not very understandable (without an IDE that can tell you what they are). In python you would use kwargs, in java you would use builder methods, is there any C# way of making a list of `false`s and `true`s less cryptic? 🤔

Reply:
Yeah, in C# you can definitely do this with keywords as well. I've done it that way in OPC-UA, this is just me being lazy.